elastic/beats
# Fix behavior of hostfs under agent

## What does this PR do?

This is a fix for : https://github.com/elastic/beats/issues/28459

Under agent, `system.hostfs` doesn't seem to get applied to individual beats. The config is passed along, but never instantiated. I've given up trying to figure out how config gets from agent to metricbeat, but this is at least a temporary fix that re-instantiates the `hostfs` value in the main system module code. We need something similar in the linux module, but that code needs enough change that I want to do that in a different PR. I'm not sure when this bug was introduced or what changed. There might be a more elegant way to fix this, but I don't know enough about how agent "reloads" beat configs to figure out what that is, so this is a bit of an awkward fix.

## Why is it important?

`system.hostfs` is broken under agent, which is bad.

## Checklist


## How to test this PR locally

This is honestly kind of a pain to test, particularly with `elastic-package`. 
- Build agent along with metricbeat
- run and enroll agent, add `/hostfs` or any junk value to the system module agent config for the system root.
- Check the logs, If the agent suddenly breaks, that means it works.
